server/core/basic_cluster.go

```diff
@@ -288,13 +289,15 @@ func (bc *BasicCluster) TakeStore(storeID uint64) *StoreInfo {
 // PreCheckPutRegion checks if the region is valid to put.
 func (bc *BasicCluster) PreCheckPutRegion(region *RegionInfo) (*RegionInfo, error) {
 	bc.RLock()
-	for _, item := range bc.Regions.GetOverlaps(region) {
-		if region.GetRegionEpoch().GetVersion() < item.GetRegionEpoch().GetVersion() {
-			bc.RUnlock()
-			return nil, ErrRegionIsStale(region.GetMeta(), item.GetMeta())
+	origin := bc.Regions.GetRegion(region.GetID())
+	if origin == nil || !bytes.Equal(origin.GetStartKey(), region.GetStartKey()) || !bytes.Equal(origin.GetEndKey(), region.GetEndKey()) {
```

Comment:
Does it improve performance when there are lots of merge or split operations (will have a lot of overlaps) since we need to compare it one more time than before?

Reply:
In response to this problem, I conducted 3 tests:
Test 1: master branch

Test 2: Update the 'if condition' like this:
```
If origin == nil ||! bytes.Equal (origin.GetStartKey (), region.GetStartKey ()) ||! bytes.Equal (origin.GetEndKey (), region.GetEndKey ()) || true {
```

Test 3: Update the 'if conditions' like this:
```
If origin == nil ||! bytes.Equal (origin.GetStartKey (), region.GetStartKey ()) ||! bytes.Equal (origin.GetEndKey (), region.GetEndKey ()) || region.GetID ()% 100 == 0 {
```
The performance of Test 2 and Test 1 is almost the same, so it is proved that adding a comparison condition in the PR has no effect on performance

In the Test 3,  there war 1 million regions, 10,000 of them must be checked by getOverlaps. This test simulates the merge or split operation of a large number of regions in the actual project. The heartbeat processing ability of Test 3 is better than Test 1 and Test 2 a lot of.

Therefore, the effectiveness of this PR can be proved.